protocol-units/settlement/mcr/contracts/test/staking/MovementStaking.t.sol

```diff
@@ -429,4 +435,125 @@ contract MovementStakingTest is Test {
         );
         staking.reward(attesters, amounts, custodians);
     }
+
+    function testBasicDelegation() public {
+        MOVEToken moveToken = new MOVEToken();
+        moveToken.initialize();
+
+        MovementStaking staking = new MovementStaking();
+        staking.initialize(moveToken);
```

Comment:
I would advise against building tests that initalize the implementation contract and not a proxy contract. This might lead to clash in functions or impredictable behavior. There has been recently a hack on deltaprime that was caused by this.
I'd be glad to redesign these tests.

Reply:
That's probably best for a separate PR as literally all of the tests use this approach.

Reply:
Can you provide some links here. It would seem to me you'd have to be doing something exotic with your use of the proxy pattern to actually have behavior vary in ways that tests like these wouldn't assert.

Reply:
Or, in the least, you would not need to check all of the logic and instead access controls.

Reply:
https://medium.com/@Delta_Prime/deltaprime-post-mortem-report-752bd60a25e6
I think it's a bit further than what I expected but I think that anything that gets us closer to the actual implementation might prevent us from running into a critical bug in production.